src/main/java/org/prebid/server/bidder/ttx/TtxBidder.java

```diff
@@ -99,90 +99,88 @@ private ExtImpTtx parseImpExt(Imp imp) {
         }
     }
 
-    private Imp updateFirstImp(Imp firstImp, String productId,
-                               String zoneId, List<BidderError> errors) {
-        final Imp.ImpBuilder modifiedFirstImp = firstImp.toBuilder();
-        modifiedFirstImp.ext(createImpExt(productId, zoneId)).build();
-
-        final Video video = firstImp.getVideo();
-        try {
-            if (video != null) {
-                modifiedFirstImp.video(updateVideo(video, productId));
-            }
-        } catch (PreBidException e) {
-            errors.add(BidderError.badInput(e.getMessage()));
-        }
-
-        return modifiedFirstImp.build();
+    private static Site updateSite(Site site, String siteId) {
+        return site == null ? null : site.toBuilder().id(siteId).build();
     }
 
-    private List<Imp> replaceFirstImp(List<Imp> imps, Imp firstImp) {
-        final List<Imp> updatedImpList = new ArrayList<>(imps);
-        updatedImpList.set(0, firstImp);
-        return updatedImpList;
+    private Imp updateImp(Imp imp, String productId, String zoneId, List<BidderError> errors) {
+
+        return imp.toBuilder()
+                .video(updateVideo(imp.getVideo(), productId, errors))
+                .ext(createImpExt(productId, zoneId))
+                .build();
     }
 
     private ObjectNode createImpExt(String productId, String zoneId) {
         final TtxImpExt ttxImpExt = TtxImpExt.of(TtxImpExtTtx.of(productId, StringUtils.stripToNull(zoneId)));
         return mapper.mapper().valueToTree(ttxImpExt);
     }
 
-    private Video updateVideo(Video video, String productId) {
+    private static Video updateVideo(Video video, String productId, List<BidderError> errors) {
+        if (video == null) {
+            return null;
+        }
         if (isZeroOrNullInteger(video.getW())
                 || isZeroOrNullInteger(video.getH())
                 || CollectionUtils.isEmpty(video.getProtocols())
                 || CollectionUtils.isEmpty(video.getMimes())
                 || CollectionUtils.isEmpty(video.getPlaybackmethod())
         ) {
-            throw new PreBidException("One or more invalid or missing video field(s) "
-                    + "w, h, protocols, mimes, playbackmethod");
+            errors.add(BidderError.badInput("One or more invalid or missing video field(s) "
+                    + "w, h, protocols, mimes, playbackmethod"));
+            return video;
```

Comment:
I also notice that they set video to null if they have any validation errors.
So we actually need to perfom ImpValidation after update. (check for banner of video)